test/unit/components/navpatientheader/index.test.js

```diff
@@ -0,0 +1,309 @@
+  
+      expect(mockTrackMetric.calledOnceWithExactly('Clicked Navbar View Data'));
+      expect(mockHistory.push.calledOnceWithExactly('/patients/1234/data')).to.be.true;
+    })
+  
+    it('Profile button links to correct page', () => {
+      const button = wrapper.find('button#navPatientHeader_profileButton').hostNodes();
+      button.simulate('click');
+  
+      expect(mockTrackMetric.calledOnceWithExactly('Clicked Navbar Name'));
+      expect(mockHistory.push.calledOnceWithExactly('/patients/1234/profile')).to.be.true;
+    })
+  
+    it('Share button links to correct page', () => {
+      const button = wrapper.find('button#navPatientHeader_shareButton').hostNodes();
+      button.simulate('click');
+  
+      expect(mockTrackMetric.calledOnceWithExactly('Clicked Navbar Share Data'));
+      expect(mockHistory.push.calledOnceWithExactly('/patients/1234/share')).to.be.true;
+    })
+  
+    it('Upload Button opens the upload dialog', () => {
+      expect(wrapper.find('.UploadLaunchOverlay').exists()).to.be.false;
+  
+      const button = wrapper.find('button#navPatientHeader_uploadButton').hostNodes();
+      button.simulate('click');
+  
+      expect(mockTrackMetric.calledOnceWithExactly('Clicked Navbar Upload Data'));
+      expect(wrapper.find('.UploadLaunchOverlay').exists()).to.be.true;
+    });
+  });
+
+  describe('button functions for clinicaian users', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      const props = { 
+        user: { roles: ['clinician'] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: true } },
+      };
+
+      wrapper = mount(
+        <BrowserRouter>
+          <NavPatientHeader {...props} />
+        </BrowserRouter>
+      );
+    })
+
+    it('View button links to correct page', () => {
+      const button = wrapper.find('button#navPatientHeader_viewDataButton').hostNodes();
+      button.simulate('click');
+  
+      expect(mockTrackMetric.calledOnceWithExactly('Clicked Navbar View Data'));
+      expect(mockHistory.push.calledOnceWithExactly('/patients/1234/data')).to.be.true;
+    })
+  
+    it('Profile button links to correct page', () => {
+      const button = wrapper.find('button#navPatientHeader_profileButton').hostNodes();
+      button.simulate('click');
+  
+      expect(mockTrackMetric.calledOnceWithExactly('Clicked Navbar Name'));
+      expect(mockHistory.push.calledOnceWithExactly('/patients/1234/profile')).to.be.true;
+    })
+  
+    it('Upload Button opens the upload dialog', () => {
+      expect(wrapper.find('.UploadLaunchOverlay').exists()).to.be.false;
+  
+      const button = wrapper.find('button#navPatientHeader_uploadButton').hostNodes();
+      button.simulate('click');
+  
+      expect(mockTrackMetric.calledOnceWithExactly('Clicked Navbar Upload Data'));
+      expect(wrapper.find('.UploadLaunchOverlay').exists()).to.be.true;
+    });
+  });
+
+  describe('back button', () => {
```

Comment:
These tests are adapted and essentially unchanged from `test/unit/components/navbar.test.js`